posthog/warehouse/api/external_data_source.py

```diff
@@ -308,7 +315,9 @@ def safely_get_queryset(self, queryset):
                     "schemas",
                     queryset=ExternalDataSchema.objects.filter(team_id=self.team_id)
                     .exclude(deleted=True)
-                    .filter(should_sync=True)
+                    .filter(
+                        Q(should_sync=True) | Q(latest_error__isnull=False)
+                    )  # OR to include schemas with errors or marked for sync
```

Comment:
@Gilbert09 This does change what is returned in a test since it includes records with non-null errors. I _think_ this is what we want since the latest_error will get nulled out when the connection is rectified.

Reply:
I guess my question here is whether we want the source to say `Error` when the errored schema is now disabled?

Reply:
Maybe we add a new visual state, like `SUSPENDED` or something like that?